All_home_works/hw9/task1.py

```diff
@@ -16,24 +16,27 @@
 
 
 def merge_sorted_files(file_list: List[Union[Path, str]]) -> Iterator:
-    num_generators = [_get_nums_from_file(file) for file in file_list]
-    numbers = [
-        (index, next(number, float("inf")))
-        for index, number in enumerate(num_generators)
-    ]
-    stop = sum(num[1] == float("inf") for num in numbers)
-    while stop < len(numbers):
-        num, stop = _get_min_number(numbers, num_generators, stop)
-        yield num
-
-
-def _get_min_number(numbers, num_generators, stop):
-    index, num = min(numbers, key=lambda x: x[1])
-    next_number = next(num_generators[index], float("inf"))
-    numbers[index] = (index, next_number)
-    if next_number == float("inf"):
-        stop += 1
-    return num, stop
+    numbers__generators = _get_numbers_and_generators(file_list)
```

Comment:
It's more common to use a single underscore in names.  `snake_case` not `snake__case`.